### PR TITLE
Don't checkpoint when max attempts exceeded

### PR DESCRIPTION
**Pull request by nsinkov**
## Changes proposed in this PR

- add option to not checkpoint after some number of job attempts have failed

## Why are we making these changes?

Checkpointing itself can cause failure. This will allow a job to complete

### CODE REVIEW

File: scheduler/src/cook/kubernetes/api.clj [hunk 1]
```diff
@@ -605,6 +605,31 @@
...
+    :mesos-container-launch-failed})
+
+(defn calculate-effective-checkpointing-config
+  "Given the job's checkpointing config, calculate the effective config. Making any adjustments such as defaults,
+  overrides, or other behavior modifications."
+  [{:keys [job/checkpoint job/instance] :as job} task-id]
+  (when checkpoint
+    (let [{:keys [default-checkpoint-config]} (config/kubernetes)
+          checkpoint (merge default-checkpoint-config (util/job-ent->checkpoint job))]
+      (if-let [max-checkpoint-attempts (:max-checkpoint-attempts checkpoint)]
```
Comment (dposada) [inline scheduler/src/cook/kubernetes/api.clj:622]: Can we destructure this in the `let` above?

File: scheduler/src/cook/kubernetes/api.clj [hunk 2]
```diff
@@ -605,6 +605,31 @@
...
+        (let [checkpoint-failure-reasons (or (:checkpoint-failure-reasons checkpoint) default-checkpoint-failure-reasons)
```
Comment (dposada) [inline scheduler/src/cook/kubernetes/api.clj:623]: Destructure?

File: scheduler/src/cook/kubernetes/api.clj (hunk 1 above)
Comment (nsinkov) [reply scheduler/src/cook/kubernetes/api.clj]: updated

File: scheduler/src/cook/kubernetes/api.clj (hunk 2 above)
Comment (nsinkov) [reply scheduler/src/cook/kubernetes/api.clj]: updated

**Comment by dposada** [comment]
Will merge when green

**Pull request merged**